### ISSUE TITLE
www.captiongenerator.com

### ISSUE THREAD

**Issue by SasquatchYuja** [ytdl-org/youtube-dl#28291]
## Checklist

- [x] I'm reporting a new site support request
- [x] I've verified that I'm running youtube-dl version **2021.02.22**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that none of provided URLs violate any copyrights
- [x] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- Single video: https://www.captiongenerator.com/128/Team-building
- Single video: https://www.captiongenerator.com/2153287/Man-finds-phone
- Single video: https://www.captiongenerator.com/2140517/Haavisto-Pietarissa
- Single video: https://www.captiongenerator.com/2153147/81-Pay

## Description

This is a website aimed at changing original captions, usually to make parodies, thus the many Downfall templates.
Some time ago if I remember well, it also allowed you to set the desired video from any Youtube URL, apparently it is now restricted to a certain set only.

The aim would be to download the video with a subtitles channel at best, else hardcoded captions.

Currently, calling `youtube-dl` results in a 403. My first move after that error is usually to delete cache, adding cookies does not seem to change anything neither, althought no authentification is required, nor captcha as far as I know.

URLs seem to match regexp 'https://www\.captiongenerator\.com/[0-9]+/[a-zA-Z0-9+-]'

```sh
youtube-dl --rm-cache-dir && youtube-dl "https://www.captiongenerator.com/128/Team-building" -v -C --cookies cookies.txt
```
```
Removing cache dir /home/user/.cache/youtube-dl ..
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['https://www.captiongenerator.com/128/Team-building', '-v', '-C', '--cookies', 'b']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.02.22
[debug] Python version 3.9.1 (CPython) - Linux-4.14.221-1-MANJARO-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.3.1, ffprobe 4.3.1, rtmpdump 2.4
[debug] Proxy map: {}
[debug] Public IP address:
[generic] Team-building: Requesting header
WARNING: Falling back on generic information extractor.
[generic] Team-building: Downloading webpage
[generic] Team-building: Extracting information
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.9/site-packages/youtube_dl/YoutubeDL.py", line 806, in wrapper
    return func(self, *args, **kwargs)
  File "/home/user/.local/lib/python3.9/site-packages/youtube_dl/YoutubeDL.py", line 827, in __extract_info
    ie_result = ie.extract(url)
  File "/home/user/.local/lib/python3.9/site-packages/youtube_dl/extractor/common.py", line 532, in extract
    ie_result = self._real_extract(url)
  File "/home/user/.local/lib/python3.9/site-packages/youtube_dl/extractor/generic.py", line 3355, in _real_extract
    self._sort_formats(entry['formats'])
  File "/home/user/.local/lib/python3.9/site-packages/youtube_dl/extractor/common.py", line 1367, in _sort_formats
    raise ExtractorError('No video formats found')
youtube_dl.utils.ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

**Comment by brlipi** [comment]
After searching about Amazon CDN's returning 403 on request I found this [question on stack overflow](https://stackoverflow.com/questions/34882125/getting-403-forbidden-when-loading-aws-cloudfront-file).
Apparently the CDN may need a specific referer to respond with the content.

So on the captiongenerator website I opened my browser's DevTools on the Network tab and looked for the mp4 video request. The request header indeed had a referer header in it:

> :authority: d34ov3vwfhhb30.cloudfront.net
> :method: GET
> :path: /Hitler+Reacts+-+No+Subtitles.mp4
> :scheme: https
> accept: */*
> accept-encoding: identity;q=1, *;q=0
> accept-language: en-US,en;q=0.9
> if-range: "e0ef75b6e838b1b43921c553b07a7713"
> range: bytes=209192-8224767
> **referer: https://www.captiongenerator.com/**
> sec-fetch-dest: video
> sec-fetch-mode: no-cors
> sec-fetch-site: cross-site
> sec-gpc: 1
> user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.72 Safari/537.36

Note: The browser I tested this uses and ad blocker, so perhaps it will take a little longer for the browser to request the video as to let the ad play first.

Using the following command:
```
youtube-dl --add-header 'Referer: https://www.captiongenerator.com/' https://d34ov3vwfhhb30.cloudfront.net/Hitler+Reacts+-+No+Subtitles.mp4
```
I was able to download the video, although without the subtitles.

The subtitles were requested by the browser from https://www.captiongenerator.com/videos/128.vtt. No 403 issues from this one, since the subtitles are stored on the website, not the cdn. Note that the vtt file has the video's ID as it's name.

And according to <https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API>:
> Web Video Text Tracks Format (WebVTT) is a format for displaying timed text tracks (such as subtitles or captions) using the \<track\> element. The primary purpose of WebVTT files is to add text overlays to a \<video\>

But, for some strange reason they don't seem to use neither one of those tags. Instead they stick it all inside a div along with other content in a "data-react-props" attribute:
```
<div data-react-class="video/show" data-react-props="{"video":{"id":128,"title":"Team building...","view_count":823,"popularity":12.562362175914883,"updated_at":"2021-03-09T19:54:32.000Z","remote_id":"t7PmzdINGZk","description":"We believe Matt is looking for women So far, he has looked here... here, and here. From what we can tell, they don&#39;t like his goatee, but beyond that, it&#39;s not clear what they like. Sign onto his dating website...","user_id":0,"status":1,"created_at":"2011-10-31T09:10:51.000Z","service":"S3","s3_url":"https://d34ov3vwfhhb30.cloudfront.net/Hitler+Reacts+-+No+Subtitles.mp4","slug":"Team-building"},"captions":[{"id":2140,"video_id":128,"text":"We believe Matt is looking for women","start":0.5,"end":3.25,"created_at":null,"updated_at":null,"template_id":null},{"id":2141,"video_id":128,"text":"So far, he has looked here...","start":4.0,"end":5.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2142,"video_id":128,"text":"here, and here.","start":5.65,"end":7.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2143,"video_id":128,"text":"From what we can tell, they don&#39;t like his goatee,","start":8.0,"end":12.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2144,"video_id":128,"text":"but beyond that, it&#39;s not clear what they like.","start":12.1,"end":15.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2145,"video_id":128,"text":"Sign onto his dating website...","start":17.5,"end":19.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2146,"video_id":128,"text":"We&#39;ll find out what they do like about him.","start":19.0,"end":21.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2147,"video_id":128,"text":"Boss..","start":24.75,"end":26.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2148,"video_id":128,"text":"They...","start":27.5,"end":28.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2149,"video_id":128,"text":"They don&#39;t like small penises either.","start":31.0,"end":33.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2150,"video_id":128,"text":"They are using toothpicks to size him up.","start":34.0,"end":36.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2151,"video_id":128,"text":"Small penises support group...stay in the room. Irish, Army boy, Tom.","start":53.0,"end":58.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2152,"video_id":128,"text":"What the hell are they thinking?","start":73.0,"end":75.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2153,"video_id":128,"text":"Using a stick to measue a flacid penis?","start":75.25,"end":77.75,"created_at":null,"updated_at":null,"template_id":null},{"id":2154,"video_id":128,"text":"And, what the fuck is it with no goatees?","start":78.75,"end":83.25,"created_at":null,"updated_at":null,"template_id":null},{"id":2155,"video_id":128,"text":"What do they like about this man?","start":85.25,"end":88.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2156,"video_id":128,"text":"His sense of humor?!?!","start":89.25,"end":91.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2157,"video_id":128,"text":"I have a small penis! I&#39;m short, and I smell.","start":91.5,"end":94.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2158,"video_id":128,"text":"I get laid, in cars, houses....this bunker too.","start":94.75,"end":97.75,"created_at":null,"updated_at":null,"template_id":null},{"id":2159,"video_id":128,"text":"I have this little &#39;stache. Do you like it?","start":97.5,"end":100.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2160,"video_id":128,"text":"Sir. Not really, Sir. It looks like a dead mouse.","start":100.5,"end":102.75,"created_at":null,"updated_at":null,"template_id":null},{"id":2161,"video_id":128,"text":"Ass-hole. Do you know how hard it is to trim this fucker?","start":102.75,"end":106.25,"created_at":null,"updated_at":null,"template_id":null},{"id":2162,"video_id":128,"text":"Sir, they are frustrated and looking for clean shaven men.","start":106.5,"end":108.8,"created_at":null,"updated_at":null,"template_id":null},{"id":2163,"video_id":128,"text":"They&#39;ll get a smooth face, but what of it? At least use a pencil.","start":108.8,"end":112.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2164,"video_id":128,"text":"O&#39;Donnell! You&#39;re from Eireann.","start":113.0,"end":114.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2165,"video_id":128,"text":"The Irish have small penises, and you do alright.","start":116.0,"end":117.75,"created_at":null,"updated_at":null,"template_id":null},{"id":2166,"video_id":128,"text":"Yes, you&#39;re not on the HottestPeopleSeeker.com web site...","start":117.75,"end":120.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2167,"video_id":128,"text":"But, you&#39;ve done alright!!!!","start":120.5,"end":123.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2168,"video_id":128,"text":"You&#39;re getting married in a few months, so you&#39;re off the market.","start":124.5,"end":128.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2169,"video_id":128,"text":"There are women outside who&#39;d fling themsleves at Berg.","start":128.75,"end":133.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2170,"video_id":128,"text":"And he can play those small guitars.","start":134.0,"end":136.75,"created_at":null,"updated_at":null,"template_id":null},{"id":2171,"video_id":128,"text":"If only we can find someone who could vouch for him...like a former lover or something. Quit stalling!!!","start":137.5,"end":141.75,"created_at":null,"updated_at":null,"template_id":null},{"id":2172,"video_id":128,"text":"She&#39;s be happy for him....I&#39;m sure of it.","start":147.0,"end":149.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2173,"video_id":128,"text":"He has Norse Viking blood in him...","start":150.5,"end":154.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2174,"video_id":128,"text":"He was in the Navy. He&#39;s a frigging warrior.","start":154.5,"end":156.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2175,"video_id":128,"text":"Too many people...","start":161.0,"end":162.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2176,"video_id":128,"text":"Too many people enhance their online profiles. Say what they think they look like.","start":163.0,"end":167.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2177,"video_id":128,"text":"Halve the time, a 6 foot blond turns out to be a 3 foot midget.","start":168.5,"end":173.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2178,"video_id":128,"text":"Dating sucks. He kills deer for food, you know?","start":174.0,"end":176.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2179,"video_id":128,"text":"You gotta get out and meet people!","start":176.5,"end":179.75,"created_at":null,"updated_at":null,"template_id":null},{"id":2180,"video_id":128,"text":"Online dating is a sham!","start":180.0,"end":182.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2181,"video_id":128,"text":"Matt likes you. Just shave the bush.","start":184.75,"end":187.5,"created_at":null,"updated_at":null,"template_id":null},{"id":2182,"video_id":128,"text":"Forget the girls. Who&#39;s done their review?","start":194.25,"end":196.25,"created_at":null,"updated_at":null,"template_id":null},{"id":2183,"video_id":128,"text":"We have to give out a review average of under 3 to each of you.","start":199.25,"end":203.25,"created_at":null,"updated_at":null,"template_id":null},{"id":2184,"video_id":128,"text":"Bonuses?","start":205.55,"end":206.75,"created_at":null,"updated_at":null,"template_id":null},{"id":2185,"video_id":128,"text":"No bonus for you, army man.","start":211.5,"end":213.75,"created_at":null,"updated_at":null,"template_id":null},{"id":2186,"video_id":128,"text":"Tom,...You did all of his work, and most of Irish&#39;s, too, I hear.","start":220.75,"end":226.25,"created_at":null,"updated_at":null,"template_id":null},{"id":2187,"video_id":128,"text":"The image, Office, Windows  7. Exchange. All Tom!","start":226.5,"end":229.0,"created_at":null,"updated_at":null,"template_id":null},{"id":2188,"video_id":128,"text":"Henry, pack your bags.","start":233.75,"end":236.0,"created_at":null,"updated_at":null,"template_id":null}],"trending":[{"id":2163917,"remote_id":"t7PmzdINGZk","title":"Ribbons and stickers and chalk oh my!","description":"Or when politicians see civil protest as criminal acts and threaten the law to silence dissent","view_count":1234,"user_id":null,"status":null,"created_at":"2021-03-08T23:31:35.000Z","updated_at":"2021-03-09T19:52:49.000Z","service":"S3","s3_url":"https://d34ov3vwfhhb30.cloudfront.net/Hitler+Reacts+-+No+Subtitles.mp4","popularity":1084.16,"slug":"Ribbons-and-stickers-and-chalk-oh-my"},{"id":2158327,"remote_id":"t7PmzdINGZk","title":"ONS muck up the sex question in the census","description":"ONS officials tell the Chief Statistician that they&#39;re being taken to court","view_count":1254,"user_id":null,"status":null,"created_at":"2021-03-03T22:08:54.000Z","updated_at":"2021-03-09T19:45:26.000Z","service":"S3","s3_url":"https://d34ov3vwfhhb30.cloudfront.net/Hitler+Reacts+-+No+Subtitles.mp4","popularity":490.428,"slug":"ONS-muck-up-the-sex-question-in-the-census"},{"id":2164287,"remote_id":"t7PmzdINGZk","title":"David Paisley reacts to #Superstraight","description":"David gets ready to deploy his penis to stop the #superstraights in their tracks. ","view_count":464,"user_id":null,"status":null,"created_at":"2021-03-09T11:32:19.000Z","updated_at":"2021-03-09T19:53:41.000Z","service":"S3","s3_url":"https://d34ov3vwfhhb30.cloudfront.net/Hitler+Reacts+-+No+Subtitles.mp4","popularity":432.308,"slug":"David-Paisley-reacts-to-Superstraight"},{"id":2158277,"remote_id":"t7PmzdINGZk","title":"La chute du Pas-de-Calais","description":"3 mars 2021 : Face à la situation sanitaire, le Président de la République décide le confinement du Pas-de-Calais pendant 4 week-end. ","view_count":1004,"user_id":null,"status":null,"created_at":"2021-03-03T21:28:18.000Z","updated_at":"2021-03-09T17:46:31.000Z","service":"S3","s3_url":"https://d34ov3vwfhhb30.cloudfront.net/Hitler+Reacts+-+No+Subtitles.mp4","popularity":267.71,"slug":"La-chute-du-Pas-de-Calais"},{"id":2159887,"remote_id":"rnLgDqDQ1YE","title":"GH fleet tactics be like","description":"Gh has been taking fleet roams into our territory but always logs off when we arrive","view_count":722,"user_id":null,"status":null,"created_at":"2021-03-04T22:17:51.000Z","updated_at":"2021-03-09T19:17:00.000Z","service":"S3","s3_url":"https://d34ov3vwfhhb30.cloudfront.net/Laughing+Man.mp4","popularity":266.193,"slug":"GH-fleet-tactics-be-like"}],"csrfToken":"C2xdp894OSMCSOtLO3unHgydlHFNK+rZy9QFvK0RyH0DLF/QCAeFmqj/NYMt7moxYgAKL/ZGqgKnAjBAF6IjwA=="}" data-hydrate="t"><div class="flex-l" itemProp="video" itemscope="" itemType="http://schema.org/VideoObject" data-reactroot=""><div style="flex:1 1 0"><div data-vjs-player="true" class="overflow-hidden br2"><video id="cg-player" class="video-js vjs-fluid vjs-big-play-centered pointer"><track srcLang="en" kind="subtitles" type="text/vtt" src="/videos/128.vtt" default="" /></video></div><div class="fr pa2 f4"><span itemProp="interactionStatistic" itemscope="" itemType="http://schema.org/InteractionCounter"><span itemProp="interactionService" itemscope="" itemID="http://www.captiongenerator.com/" itemType="http://schema.org/Website"><meta itemProp="name" content="Caption Generator" /></span><link itemProp="interactionType" href="https://schema.org/WatchAction" /><span itemProp="userInteractionCount">823</span></span> <!-- -->views</div><div><div itemProp="name" class="f3 lh-title mt1">Team building...</div><meta itemProp="thumbnail" content="https://img.youtube.com/vi/t7PmzdINGZk/hqdefault.jpg" /><meta itemProp="thumbnailUrl" content="https://img.youtube.com/vi/t7PmzdINGZk/hqdefault.jpg" /><meta itemProp="embedUrl" content="https://d34ov3vwfhhb30.cloudfront.net/Hitler+Reacts+-+No+Subtitles.mp4" /><meta itemProp="playerType" content="mp4" /><meta itemProp="width" content="480" /><meta itemProp="height" content="320" /><meta itemProp="caption" content="https://www.captiongenerator.com/videos/128.vtt" /><meta itemProp="uploadDate" content="2011-10-31T09:10:51.000Z" /><div itemProp="description" class="mv2">We believe Matt is looking for women So far, he has looked here... here, and here. From what we can tell, they don&#x27;t like his goatee, but beyond that, it&#x27;s not clear what they like. Sign onto his dating website...</div></div><h4>Share</h4><div class="addthis_toolbox addthis_default_style addthis_32x32_style"><a class="addthis_button_preferred_1"></a><a class="addthis_button_preferred_2"></a><a class="addthis_button_preferred_3"></a><a class="addthis_button_preferred_4"></a><a class="addthis_button_compact"></a><a class="addthis_counter addthis_bubble_style"></a></div><input class="w-90 pa2 ba br2" readonly="" value="https://captiongenerator.com/128/Team-building" /><div id="disqus_thread"></div><form method="POST" class="pt4 f5 w-90"><h4><a href="#">Report this video</a></h4></form><h4>Captions</h4><div itemProp="transcript" class="h5 overflow-y-auto mv3"><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:00<!-- --> - <!-- -->00:03</span>We believe Matt is looking for women</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:04<!-- --> - <!-- -->00:05</span>So far, he has looked here...</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:05<!-- --> - <!-- -->00:07</span>here, and here.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:08<!-- --> - <!-- -->00:12</span>From what we can tell, they don&#x27;t like his goatee,</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:12<!-- --> - <!-- -->00:15</span>but beyond that, it&#x27;s not clear what they like.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:17<!-- --> - <!-- -->00:19</span>Sign onto his dating website...</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:19<!-- --> - <!-- -->00:21</span>We&#x27;ll find out what they do like about him.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:24<!-- --> - <!-- -->00:26</span>Boss..</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:27<!-- --> - <!-- -->00:28</span>They...</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:31<!-- --> - <!-- -->00:33</span>They don&#x27;t like small penises either.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:34<!-- --> - <!-- -->00:36</span>They are using toothpicks to size him up.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">00:53<!-- --> - <!-- -->00:58</span>Small penises support group...stay in the room. Irish, Army boy, Tom.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:13<!-- --> - <!-- -->01:15</span>What the hell are they thinking?</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:15<!-- --> - <!-- -->01:17</span>Using a stick to measue a flacid penis?</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:18<!-- --> - <!-- -->01:23</span>And, what the fuck is it with no goatees?</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:25<!-- --> - <!-- -->01:28</span>What do they like about this man?</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:29<!-- --> - <!-- -->01:31</span>His sense of humor?!?!</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:31<!-- --> - <!-- -->01:34</span>I have a small penis! I&#x27;m short, and I smell.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:34<!-- --> - <!-- -->01:37</span>I get laid, in cars, houses....this bunker too.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:37<!-- --> - <!-- -->01:40</span>I have this little &#x27;stache. Do you like it?</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:40<!-- --> - <!-- -->01:42</span>Sir. Not really, Sir. It looks like a dead mouse.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:42<!-- --> - <!-- -->01:46</span>Ass-hole. Do you know how hard it is to trim this fucker?</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:46<!-- --> - <!-- -->01:48</span>Sir, they are frustrated and looking for clean shaven men.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:48<!-- --> - <!-- -->01:52</span>They&#x27;ll get a smooth face, but what of it? At least use a pencil.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:53<!-- --> - <!-- -->01:54</span>O&#x27;Donnell! You&#x27;re from Eireann.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:56<!-- --> - <!-- -->01:57</span>The Irish have small penises, and you do alright.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">01:57<!-- --> - <!-- -->02:00</span>Yes, you&#x27;re not on the HottestPeopleSeeker.com web site...</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:00<!-- --> - <!-- -->02:03</span>But, you&#x27;ve done alright!!!!</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:04<!-- --> - <!-- -->02:08</span>You&#x27;re getting married in a few months, so you&#x27;re off the market.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:08<!-- --> - <!-- -->02:13</span>There are women outside who&#x27;d fling themsleves at Berg.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:14<!-- --> - <!-- -->02:16</span>And he can play those small guitars.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:17<!-- --> - <!-- -->02:21</span>If only we can find someone who could vouch for him...like a former lover or something. Quit stalling!!!</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:27<!-- --> - <!-- -->02:29</span>She&#x27;s be happy for him....I&#x27;m sure of it.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:30<!-- --> - <!-- -->02:34</span>He has Norse Viking blood in him...</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:34<!-- --> - <!-- ->02:36</span>He was in the Navy. He&#x27;s a frigging warrior.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:41<!-- --> - <!-- -->02:42</span>Too many people...</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:43<!-- --> - <!-- -->02:47</span>Too many people enhance their online profiles. Say what they think they look like.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:48<!-- --> - <!-- -->02:53</span>Halve the time, a 6 foot blond turns out to be a 3 foot midget.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:54<!-- --> - <!-- -->02:56</span>Dating sucks. He kills deer for food, you know?</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">02:56<!-- --> - <!-- -->02:59</span>You gotta get out and meet people!</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">03:00<!-- --> - <!-- -->03:02</span>Online dating is a sham!</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">03:04<!-- --> - <!-- -->03:07</span>Matt likes you. Just shave the bush.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">03:14<!-- --> - <!-- -->03:16</span>Forget the girls. Who&#x27;s done their review?</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">03:19<!-- --> - <!-- -->03:23</span>We have to give out a review average of under 3 to each of you.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">03:25<!-- --> - <!-- -->03:26</span>Bonuses?</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">03:31<!-- --> - <!-- -->03:33</span>No bonus for you, army man.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">03:40<!-- --> - <!-- -->03:46</span>Tom,...You did all of his work, and most of Irish&#x27;s, too, I hear.</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">03:46<!-- --> - <!-- -->03:49</span>The image, Office, Windows 7. Exchange. All Tom!</div><div class="f6 pv1 hover-bg-yellow bg-animate"><span class="dib w4">03:53<!-- --> - <!-- -->03:56</span>Henry, pack your bags.</div></div></div><div class="mh2"><ins class="adsbygoogle" style="display:block" data-ad-client="ca-pub-0329427363644241" data-ad-slot="8659892150" data-ad-format="auto" data-full-width-responsive="true"></ins><h3 class="mt1">Make your own video</h3><a class="link bg-green white pa3 br2 dib grow f4 b w-100 tc" href="/videos/template/128">Create a video from this template</a><h3>Trending</h3><div class="mw6-l flex flex-wrap flex-column-l"><a href="/2163917/Ribbons-and-stickers-and-chalk-oh-my" class="dib link ma2 grow mt0 relative" style="width:256px"><div class="overflow-hidden fl mr2 br2 relative" style="width:256px;height:128px"><img src="//img.youtube.com/vi/t7PmzdINGZk/hqdefault.jpg" alt="Ribbons and stickers and chalk oh my!" style="margin-top:25%;margin-left:50%;transform:translate(-50%, -50%)" /></div><div class="absolute top-0 text-shadow pa2 white f6"><div class="f5 b mv1">Ribbons and stickers and chalk oh my!</div><div class="mb1">20 hours ago</div><div class="f6"><span class="b">1234</span> <!-- -->views</div></div></a><a href="/2158327/ONS-muck-up-the-sex-question-in-the-census" class="dib link ma2 grow mt0 relative" style="width:256px"><div class="overflow-hidden fl mr2 br2 relative" style="width:256px;height:128px"><img src="//img.youtube.com/vi/t7PmzdINGZk/hqdefault.jpg" alt="ONS muck up the sex question in the census" style="margin-top:25%;margin-left:50%;transform:translate(-50%, -50%)" /></div><div class="absolute top-0 text-shadow pa2 white f6"><div class="f5 b mv1">ONS muck up the sex question in the census</div><div class="mb1">6 days ago</div><div class="f6"><span class="b">1254</span> <!-- -->views</div></div></a><a href="/2164287/David-Paisley-reacts-to-Superstraight" class="dib link ma2 grow mt0 relative" style="width:256px"><div class="overflow-hidden fl mr2 br2 relative" style="width:256px;height:128px"><img src="//img.youtube.com/vi/t7PmzdINGZk/hqdefault.jpg" alt="David Paisley reacts to #Superstraight" style="margin-top:25%;margin-left:50%;transform:translate(-50%, -50%)" /></div><div class="absolute top-0 text-shadow pa2 white f6"><div class="f5 b mv1">David Paisley reacts to #Superstraight</div><div class="mb1">8 hours ago</div><div class="f6"><span class="b">464</span> <!-- -->views</div></div></a><a href="/2158277/La-chute-du-Pas-de-Calais" class="dib link ma2 grow mt0 relative" style="width:256px"><div class="overflow-hidden fl mr2 br2 relative" style="width:256px;height:128px"><img src="//img.youtube.com/vi/t7PmzdINGZk/hqdefault.jpg" alt="La chute du Pas-de-Calais" style="margin-top:25%;margin-left:50%;transform:translate(-50%, -50%)" /></div><div class="absolute top-0 text-shadow pa2 white f6"><div class="f5 b mv1">La chute du Pas-de-Calais</div><div class="mb1">6 days ago</div><div class="f6"><span class="b">1004</span> <!-- -->views</div></div></a><a href="/2159887/GH-fleet-tactics-be-like" class="dib link ma2 grow mt0 relative" style="width:256px"><div class="overflow-hidden fl mr2 br2 relative" style="width:256px;height:128px"><img src="//img.youtube.com/vi/rnLgDqDQ1YE/hqdefault.jpg" alt="GH fleet tactics be like" style="margin-top:25%;margin-left:50%;transform:translate(-50%, -50%)" /></div><div class="absolute top-0 text-shadow pa2 white f6"><div class="f5 b mv1">GH fleet tactics be like</div><div class="mb1">5 days ago</div><div class="f6"><span class="b">722</span> <!-- -->views</div></div></a></div></div></div></div>-
```
I briefly looked into putting .mp4 and .vtt together in one file. [There's a website which puts them together](https://www.happyscribe.com/subtitle-tools/add-vtt-to-mp4).

I tested for the "Team building" video and it worked!
![Comparison](https://user-images.githubusercontent.com/43282619/110536381-28d59080-8100-11eb-9252-6e41ac1540ef.png)

The subtitles lose their black background. I don't know why that is. Perhaps happyscribe converts it into another format (maybe .srt) before merging it with the video.

In conclusion, I think you will need to make a script that uses multiple tools together with youtube-dl to reach your goal. Especially if you wish to merge subtitles and video without relying on an online tool for it.
